ambry-store/src/test/java/com.github.ambry.store/StorageManagerTest.java

```diff
@@ -83,22 +88,15 @@ public void mountPathNotFoundTest() throws Exception {
         downReplicaCount++;
       }
     }
-    deleteDirectory(new File(mountPathToDelete));
-    StorageManager storageManager = createAndStartStoreManager(replicas, metricRegistry);
+    StorageManager storageManager = createStorageManager(replicas, metricRegistry);
+    Utils.deleteFileOrDirectory(new File(mountPathToDelete));
```

Comment:
This was because the DSA constructor could throw an exception if the reserve directory was invalid or could not be created. I realized this can prevent the node from starting up if one disk manager fails, so I changed the behavior of DSA. Now, it will make its best effort in the constructor to inventory the reserve pool if it already exists, but will not throw an exception if it does not succeed. If it fails, or the directory does not yet exist, `initializePool()` can create the directory/inventory. This stage *can* throw an exception to signify that the disk is unusable. `free`/`allocate` can handle the "not inventoried" state by avoiding the pool entirely.